File: RyanDahl/apache-example.js
var http = require('http');
var s  = http.createServer(function(req,res){
	console.log("serving to "+req.connection.remoteAddress.slice(7));
	res.writeHead(200,{"content-type":"text/plain"});
	res.write("hello\n");
	/*setTimeout(function(){
		res.end("world\n");
	},2000);*/
	res.end("world\n");
});
s.listen(8000);
console.log("serving at localhost:8000");
//use ab -n 100 -c 100 http://127.0.0.1:8000/ for tests

File: RyanDahl/apache-example.py
from socket import *
import thread
def server(connection):
    connection.send('HTTP/1.0 200 OK\n')
    connection.send('Content-Type: text/html\n')
    connection.send("""
        <html>
        <body>
        <h1>Hello World</h1> this is my server!
        </body>
        </html>
    """)
    connection.sendall()
    connection.close()
port=8000
mysock=socket(AF_INET,SOCK_STREAM)
mysock.bind(('127.0.0.1',port))
mysock.listen(1000)
while True:
    conn,addr = mysock.accept()
    thread.start_new_thread(server,(conn,))
